### ISSUE TITLE
Some characters in romanian UTF-8 subtitles are displayed as empty boxes

### ISSUE THREAD

**Issue by zelea2** [xbmc/xbmc#18867]
### Describe the bug
Romanian subtitles encoded as ISO-8859-2 or ISO-8859-16 are displayed fine.
Romanian subtitles encoded as UTF-8 have the ă Ă â Â î Î displayed correctly (when "Central Europe (ISO)" is selected) while the ş Ş ţ Ţ are displayed as empty boxes.
Other languages like Albanian, Croatian, Hungarian, Italian, Polish, Slovene might be affected

## Possible Fix
<pre>
--- utils/CharsetConverter.cpp.old      2020-10-23 22:43:37.000000000 +0100
+++ utils/CharsetConverter.cpp  2020-11-26 02:33:16.889017569 +0000
@@ -535,6 +535,7 @@
   , { "ISO-8859-7", "Greek (ISO)" }
   , { "ISO-8859-8", "Hebrew (ISO)" }
   , { "ISO-8859-9", "Turkish (ISO)" }
+  , { "ISO-8859-16", "South-Eastern Europe (ISO)" }
   , { "CP1250", "Central Europe (Windows)" }
   , { "CP1251", "Cyrillic (Windows)" }
   , { "CP1252", "Western Europe (Windows)" }
</pre>

### To Reproduce

Take any romanian subtitle (.srt) with UTF-8 encoding and enable subtitles.
To emphasise this problem try to use the iconv utility in linux:

$ iconv --output=out.srt --from-code=UTF-8 --to-code=ISO-8859-2 romanian-utf-8.srt
iconv: illegal input sequence at position 36

$ iconv --output=out.srt --from-code=UTF-8 --to-code=ISO-8859-16 romanian-utf-8.srt
the second command works fine

### Your Environment
Used Operating system:
 - [ ] Linux

 - Operating system version/name:
 - Kodi version:
kodi v18.9

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by enen92** [comment]
Can you share a sample sub file? Thanks

**Comment by DaVukovic** [comment]
Does it work any better if you set the skin font to "arial based"?

Please confirm it working or get us a sample to test with. Otherwise we might be in need to close this issue.

**Comment by zelea2** [comment]
My proposed fix is not working. Setting the skin font to  "arial based" has no effect. 
This issue could also be related to the fact that these romanian characters "ŞŢşţ" can be encoded in two ways as explained here: https://en.wikipedia.org/wiki/%C5%9E
So the 'Ş' is visually indistinguishable if encoded as (S-comma) or as (S-cedilla)

**Comment by zelea2** [comment]
I am also attaching a sample of romanian subs with this issue.
[example_subs.zip](https://github.com/xbmc/xbmc/files/5760051/example_subs.zip)

**Comment by zelea2** [comment]
After 3 and a half months no one bother to look at this bug and it is still marked as "Triage: Needed"
Well done guys.

Meanwhile I've solved it. The problem is that the provided font doesn't have the unicode glyphs for the some romanian characters. All I had to do is copy the following missing chars in arial.ttf:
U+015E -> U+0218
U+015F -> U+0219
U+0162 -> U+021A
U+0163 -> U+021B

I've done this very easily in linux using the fontforge tool.
[ If someone wants to try this you can get the modified arial.ttf from here: http://flop.vajn.icu:8765/kodi-Fonts/
I've also stripped all the chinese, japanese, arabic and hebrew fonts which I've never used and reduced the
size of the fonts from 22MB to 0.5MB but this is just for my local install ]

So in conclusion this issue is solved just by modifing the arial.tff font (copy 4 chars)
The above patch of the CharsetConverter.cpp file is also useful if you want to force Latin 10 (ISO-8859-16) encodings but be aware that recompiling kodi takes several hours.

I will leave this issue open until some maintainer fixes it in the main tree.

**Comment by fuzzard** [comment]
Feel free to PR your fix. This is open source software, the best way to get something fixed is to do it yourself. I'm sure others would welcome your improvement

**Comment by zelea2** [comment]
**Followup:** I've just realised that this is not even a kodi problem, it is debian's problem.

Debian replaces the arial.ttf original font which is 4.6MB is size with its own which is 22.7MB in size.
Only the bigger font provided by debian has the above mentioned problem.

**Comment by fuzzard** [comment]
Good to know, thanks for the followup.
